### DISCUSSION TITLE
Set up maximum duration of experiment

### DISCUSSION THREAD

**Discussion by NicoletaPrutean** [jspsych/jsPsych#983] [Q&A] [answered]
Hello, 
I want to set up a maximum duration for my experiment. 
I could come up with this solution, but it isn't an "absolute clock" (i.e., it works only if the subject leaves and then restarts the exp after max_duration has passed).  Is there a way to stop the experiment even if the participant is inactive for the entire max_duration?

Thanks.

```
jsPsych.init({
   timeline: [timeline],
   on_data_update: function (data) {
      var max_duration = 70
      var duration = (jsPsych.totalTime()/1000)/60
      if (duration > max_duration) {
          jsPsych.endExperiment('message')
      }
   }
});
```

**Comment by jodeleeuw** [comment, marked as answer]
Hi @NicoletaPrutean !

I set up a demo of this using [cognition.run](https://www.cognition.run). Here's the sample experiment: 

https://x6zzyu3dfo.cognition.run/

(It's OK to press the button. Either way the experiment should end after 3 seconds.)

Here's the code.

```js
var trial = {
  type: 'html-button-response',
  stimulus: '<p>The experiment should end after 3 seconds.</p>',
  choices: ['Don\'t press the button!'],
  on_load: function(){
    setTimeout(function(){
      jsPsych.endExperiment('The experiment is over.')
    }, 3000)
  }
}

var trial2 = {
  type: 'html-keyboard-response',
  stimulus: '<p>Oh no.</p><p>You pressed the button.</p>',
  choices: jsPsych.NO_KEYS
}

jsPsych.init({
  timeline: [trial,trial2]
})
```

The main idea is to use `setTimeout` to schedule a call to `jsPsych.endExperiment()` at some specified time. I put the scheduling call inside the on_load method of the first trial in the experiment so that the timer starts approximately when the first screen of the experiment appears.

**Reply by NicoletaPrutean** [reply]
Yes, it works, thanks!
